dali/pipeline/data/types.cu

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/data/types.h"
+
+namespace dali {
+namespace detail {
+
+__global__ void CopyKernel(uint8_t *dst, const uint8_t *src, int64_t n) {
+  for (int64_t i = blockIdx.x * blockDim.x + threadIdx.x; i < n; i += blockDim.x * gridDim.x) {
+    dst[i] = src[i];
+  }
+}
+
+void LaunchCopyKernel(void *dst, const void *src, int64_t nbytes, cudaStream_t stream) {
+  size_t block = cuda_min(nbytes, 1024l);
+  size_t grid = std::min(32l, div_ceil(nbytes, block));
```

Comment:
1. AFAIK dim3 uses `unsigned int` as element, so size_t is not necessary.
2. Don't use lowercase `l` for long numbers - looks too much like `1`.
3. `long` is not necessarily a synonym for int64_t
```suggestion
  unsigned block = std::min<int64_t>(nbytes, 1024);
  unsigned grid = std::min<int64_t>(32, div_ceil(nbytes, block));
```

Reply:
done